src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/analysers/JavaASTAnalyser.java

```diff
@@ -511,6 +517,9 @@ private void possiblyAddNavigationLink(ReviewToken token, Node node) {
      **********************************************************************************************/
 
     private void tokeniseMavenPom(Pom mavenPom) {
+        if (mavenPom == null || !mavenPom.isPomFileReal()) {
+            return;
+        }
```

Comment:
If a Maven POM is missing, that is a pretty big deal from an APIView standpoint. We should raise a diagnostic to say that a POM file is missing. Of course, in your diff case that is fine, because the NoOpDiagnostics will just ignore all diagnostics reported.